zeus/tests/test_poll_views.py

```diff
@@ -0,0 +1,164 @@
+from django.test import TestCase
+
+import datetime
+import os
+
+from helios.models import Election, Poll, Voter
+from heliosauth.models import User
+from zeus.models import Institution
+from zeus.tests.utils import SetUpAdminAndClientMixin
+
+
+class TestElectionBase(SetUpAdminAndClientMixin, TestCase):
+
+    def setUp(self):
+        super(TestElectionBase, self).setUp()
+        self.local_verbose = os.environ.get('ZEUS_TESTS_VERBOSE', None)
+        self.celebration = (
+            " _________ ___  __\n"
+            "|\   __  \|\  \|\  \\\n"
+            "\ \  \ \  \ \  \/  /_\n"
+            " \ \  \ \  \ \   ___ \\\n"
+            "  \ \  \_\  \ \  \\\ \ \\ \n"
+            "   \ \_______\ \__\\\ \_\\\n"
+            "    \|_______|\|__| \|_|\n"
+            )
+
+    def verbose(self, message):
+        if self.local_verbose:
+            print message
+
+
+    def get_institution(self, **kwargs):
+        '''
+        Create, or retrieve an instituion. For testing purposes we only need one, so
+        this instance is effectively a singleton.
+        '''
+        institution, _ = Institution.objects.get_or_create(**kwargs)
+        return institution
+
+    def get_election(self, **kwargs):
+        self.c.post(self.locations['login'], self.login_data)
+        institution = self.get_institution()
+        election, _ = Election.objects.get_or_create(institution=institution, **kwargs)
+        return election
+
+    def get_messages_from_response(self, response):
+        messages = []
+        for item in response.context['messages']:
+            messages.append(unicode(item))
+        return messages
+
+    def create_poll(self, election, **kwargs):
+        self.c.post(self.locations['login'], self.login_data)
+        poll, _ = Poll.objects.get_or_create(election=election, **kwargs)
+        self.p_uuids = []
+        for poll in election.polls.all():
+            self.p_uuids.append(poll.uuid)
+        return poll
+
+    def get_voters_file(self):
+        counter = 0
+        voter_files = {}
+        for p_uuid in self.p_uuids:
+            fname = '/tmp/random_voters%s.csv' % counter
+            voter_files[p_uuid] = fname
+            fp = file(fname, 'w')
```

Comment:
Just a proposal - in python we have the [`with` keyword that allows us for much easier work with files](https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects).

Maybe we can use it instead of more manual(you need to remember about opened files and `finally` closing them) `file.open/close` way?

Reply:
Fair point, there's a lot of these `file()` calls throughout the codebase. Maybe a separate branch?

`file` is removed in Python 3 so we'll have to touch them anyway :)